pkg/controller/realm/select.go

```diff
@@ -23,44 +23,60 @@ import (
 
 func (c *Controller) HandleSelect() http.Handler {
 	type FormData struct {
-		Realm int `form:"realm,required"`
+		RealmID uint `form:"realm,required"`
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx := r.Context()
 		flash := flash.FromContext(w, r)
 
+		session, err := c.sessions.Get(r, "session")
+		if err != nil {
+			c.logger.Errorw("failed to get session", "error", err)
+			flash.Error("Internal error, you have been logged out.")
+			http.Redirect(w, r, "/signout", http.StatusSeeOther)
+			return
+		}
+
 		user := controller.UserFromContext(ctx)
 		if user == nil {
 			flash.Error("Unauthorized.")
 			http.Redirect(w, r, "/signout", http.StatusSeeOther)
 			return
 		}
 
-		realm := controller.RealmFromContext(ctx)
```

Comment:
I'm not entirely sure how this ever worked before? This is the page that sets the realm's context, but it required a context to be set.

Reply:
RealmFromContext returned nil if one wasn't set - so this worked fine

Reply:
ah - no this got changed

I had is reading the cookie directly.